refactor `ovirtapi` to separate concerns

sjd78 opened the pull request:
`ovirtapi` has been split to have transport, data transform and API interactions in separate files:

 - HTTP transport functions are in `ovirtapi/transport.js`
 - data transform functions (from API to internal structures and back again) are in `ovirtapi/transform.js`
   - convertInt() in API transform for some values that
     transport as strings
 - API interaction functions are in `ovirtapi/index.js`
 - `ovirtapi/index.js` presents the save facade as the previous `ovirtapi.js`
 - create `ovirtapi` types, initially applied to the transform
   functions
   - each transform object has 2 corresponding types:
      - Api_Type to define the type in and out of the REST API
      - _Type to define the type within the application/redux
   - Most of the types are stubbed to `Object` so the names can be used and the actual contents can be defined from the REST API definitions and transform functions at a later time.

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+// (only for a Storage Domain type === 'iso')
+const StorageDomainFile = {
+  toInternal ({ file }: { file: Object }): Object {
```
`InternalStorageDomainFileType`

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+const Cluster = {
+  toInternal ({ cluster }: { cluster: Object }): Object {
```
`InternalClusterType`

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+const Nic = {
+  toInternal ({ nic }: { nic: Object }): Object {
```
`InternalNicType`

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+    }
+  },
+
+  toApi ({ nic }: { nic: Object }): Object {
```
`ApiNicType`

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+const VNicProfile = {
+  toInternal ({ vnicProfile }: { vnicProfile: Object }): Object {
```
`InternalVNicProfileType`

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+const Network = {
+  toInternal ({ network }: { network: Object }): Object {
```
`InternalNetworkType`

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+const Host = {
+  toInternal ({ host }: { host: Object }): Object {
```
Same.

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+const OS = {
+  toInternal ({ os }: { os: Object }): Object {
```
Same

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+const Icon = {
+  toInternal ({ icon }: { icon: Object }): Object {
```
Same.

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+const SSHKey = {
+  toInternal ({ sshKey }: { sshKey: Object }): Object {
```
Same.

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+const VmConsoles = {
+  toInternal ({ consoles }: { consoles: Object }): Array<Object> {
```
Same.

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+const VmSessions = {
+  toInternal ({ sessions }: { sessions: Object }): Object {
```
Same.

bond95 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+  sshAuthorizedKeys: string
+}
+const CloudInit = {
+  toInternal ({ vm }: { vm: Object }): CloudInitInternalType {
```
Maybe `InternalCloudInitType`?

sjd78 commented on src/ovirtapi/transform.js:
```diff
@@ -0,0 +1,615 @@
+//
+//
+type ApiVmType = Object
+type InternalVmType = Object
```
I totally agree!  I just didn't want to take the time to do that yet.  

What I can do now is put the stub types in place in another file and leave most of the as `Object` for now.  That way, the types can be filled out later as time allows.

bond95 replied:
You can actually write full type for VM model as the most important, and others in next time.

sjd78 commented:
@bond95 - I updated the PR so all of the transform functions have type names coming from __src/ovirtapi/types.js__.  I didn't add specific data type definitions beyond `Object` for most of the types.

Also, I dropped the "Internal" prefix from type names.  So "internal" types have no prefix while the REST/API types have "Api" prefixes.  I think it looks better that way.

gregsheremeta commented:
Don't merge yet. I need to release 1.4.1 today first.

gregsheremeta commented:
new commits since approval, so needs re-review. @bond95

sjd78 commented:
Rebased and squashed.  No code change since @bond95's review.

sjd78 commented:
@gregsheremeta - This PR is reviewed and ready to merge. There are no overlaps with #654 or #660.
